src/mapgen_functions.cpp

```diff
@@ -1767,74 +1778,78 @@ void mapgen_parking_lot(map *m, oter_id, mapgendata dat, int turn, float)
+        int roll = rng(1, 100);
+        if (roll <= 5) { //specials
+            int ra = rng(1, 100);
+            if (ra <= 3) {
+                veh_type = "military_cargo_truck";
+            } else if (ra <= 10) {
+                veh_type = "bubble_car";
+            } else if (ra <= 15) {
+                veh_type = "rv";
+            } else if (ra <= 20) {
+                veh_type = "schoolbus";
+            } else if (ra <= 40) {
+                veh_type = "fire_truck";
+            }else if (ra <= 60) {
+                veh_type = "policecar";
+            }else {
+                veh_type = "quad_bike";
+            }
+        } else if (roll <= 15) { //commercial
+            int rb = rng(1, 100);
+            if (rb <= 25) {
+                veh_type = "truck_trailer";
+            } else if (rb <= 35) {
+                veh_type = "semi_truck";
+            } else if (rb <= 50) {
+                veh_type = "cube_van";
             } else {
-                veh_type = "beetle";
-            }
-        } else if (rc <= 50) {
-            veh_type = "car";
-        } else if (rc <= 60) {
-            veh_type = "electric_car";
-        } else if (rc <= 65) {
-            veh_type = "hippie_van";
-        } else if (rc <= 73) {
-            veh_type = "bicycle";
-        } else if (rc <= 75) {
-            veh_type = "rara_x"; //The Solar Car Toyota RaRa X
-        } else if (rc <= 77) {
-            veh_type = "unicycle";
-        } else if (rc <= 82) {
-            veh_type = "bicycle_electric";
-        } else if (rc <= 90) {
-            veh_type = "motorcycle";
+                veh_type = "flatbed_truck";
+            }
+        } else if (roll < 50) { //commons
+            int rc = rng(1, 100);
+            if (rc <= 4) {
+                veh_type = "golf_cart";
+            } else if (rc <= 11) {
+                veh_type = "scooter";
+            } else if (rc <= 21) {
+                int rd = rng(1, 100);
+                if(rd <= 50) {
+                    veh_type = "car_mini";
+                } else {
+                    veh_type = "beetle";
+                }
+            } else if (rc <= 50) {
+                veh_type = "car";
+            } else if (rc <= 60) {
+                veh_type = "electric_car";
+            } else if (rc <= 65) {
+                veh_type = "hippie_van";
+            } else if (rc <= 73) {
+                veh_type = "bicycle";
+            } else if (rc <= 75) {
+                veh_type = "rara_x"; //The Solar Car Toyota RaRa X
+            } else if (rc <= 77) {
+                veh_type = "unicycle";
+            } else if (rc <= 82) {
+                veh_type = "bicycle_electric";
+            } else if (rc <= 90) {
+                veh_type = "motorcycle";
+            } else {
+                veh_type = "motorcycle_sidecart";
+            }
         } else {
-            veh_type = "motorcycle_sidecart";
+            veh_type = "shopping_cart";
```

Comment:
I'm open to having a slightly smaller chance of a shopping cart (currently 50% as I read this) if people want more vehicles available from parking lots.  ;-)